front/lib/api/oauth.ts

```diff
@@ -439,16 +444,23 @@ const PROVIDER_STRATEGIES: Record<
         if (!isManaged(dataSource)) {
           return null;
         }
-        const augmentedDataSource =
-          await augmentDataSourceWithConnectorDetails(dataSource);
-        if (!augmentedDataSource.connector) {
+
+        const connectorsAPI = new ConnectorsAPI(
+          config.getConnectorsAPIConfig(),
+          logger
+        );
```

Comment:
replacing the augmentDataSourceWithConnectorDetails by connectorsApi to get the ConnectionType